src/concat_str.rs

```diff
@@ -0,0 +1,69 @@
+#[cfg(test)]
+mod unit_tests;
+
+use core::cmp::min;
+use core::{
+    fmt::{Arguments as FmtArguments, Error as FmtError, Result as FmtResult, Write as FmtWrite},
+    str::from_utf8_unchecked,
+};
+
+// Create a structure to allow concatenation of `str`s in a `#[no_std]` environment
+#[derive(Debug)]
+pub struct ConcatStr<'b> {
```

Comment:
Why this module is here? It doesn't look like a part of the HAL

Reply:
It is needed to concatenate `Display` of `Error`s.